sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffleCoalesceExec.scala

```diff
@@ -147,7 +167,14 @@ object GpuShuffleCoalesceUtils {
     } else {
       hostIter
     }
-    new GpuShuffleCoalesceIterator(maybeBufferedIter, dataTypes, metricsMap)
+    if (readOption.useAsync) {
+      new GpuShuffleAsyncCoalesceIterator(maybeBufferedIter, dataTypes, targetSize,
+        outBatchesMetric, outRowsMetric, metricsMap(ASYNC_READ_TIME), opTimeMetric,
+        readThrottlingMetric)
+    } else {
+      new GpuShuffleCoalesceIterator(maybeBufferedIter, dataTypes, outBatchesMetric,
+        outRowsMetric, metricsMap(SYNC_READ_TIME), opTimeMetric)
+    }
```

Comment:
Nit. I wish we could merge the two coalesce iterators into one or have the async iterator subclass from the regular one?